src/AWS/Lambda/S3Event.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards   #-}
+
+module AWS.Lambda.S3Event
+  (
+    S3Event(..)
+  ) where
+
+import Data.Aeson
+import Data.HashMap.Strict
+import Data.Text (Text)
+import GHC.Generics
+
+data S3 a =
+  S3 {
+    s3SchemaVersion :: Text,
+    configurationId :: Text,
+    bucket          :: Bucket a,
+    s3Object        :: S3Object a
+  } deriving (Show)
+
+instance FromJSON a => FromJSON (S3 a) where
+  parseJSON (Object v) = S3
+    <$> v .: "s3SchemaVersion"
+    <*> v .: "configurationId"
+    <*> v .: "bucket"
+    <*> ((v .: "object") >>= parseJSON)
+
+instance ToJSON a => ToJSON (S3 a) where
+  toJSON S3{..} =
+    object [
+        "s3SchemaVersion" .= s3SchemaVersion
+      , "configurationId" .= configurationId
+      , "bucket" .= bucket
+      , "object" .= s3Object
+    ]
+
+data Bucket a =
```

Comment:
I don't think you need the generic parameter `a` on bucket

Reply:
Good catch

Reply:
Oh cool will update it now. Don't think it's needed from S3Object either.